homeassistant/components/template/sensor.py

```diff
@@ -176,7 +192,9 @@ async def async_setup_platform(
 
     if "coordinator" in discovery_info:
         async_add_entities(
-            TriggerSensorEntity(hass, discovery_info["coordinator"], config)
+            TriggerSensorRestoreEntity(hass, discovery_info["coordinator"], config)
+            if config.get(CONF_RESTORE, False)
```

Comment:
I think restoring trigger-based sensors should be the default

Reply:
I did not want to make it as a default or configurable as not to change existing behavior. However, if core team agrees to make this default behavior then I will certainly adjust. :-)

Reply:
Keeping default as disabled so not to change existing behavior as that might then be considered a breaking change.